File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private int ans = -1;
    public int kthSmallest(TreeNode root, int k) {
        
        bfs(root, new int[]{k});
        return ans;
    }

    private void bfs(TreeNode node, int[] k) {
        if (node == null)
            return;
        
        bfs(node.left, k);
        k[0]--;
        if (k[0] == 0) {
            ans = node.val;
            return;
        }
        bfs(node.right, k);
    }
}

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-1.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    postorderTraversal(root: TreeNode | null): number[] {
        let res = []

        const traverse = (node: TreeNode | null) => {
            if (node == null) return
            traverse(node.left)
            traverse(node.right)
            res.push(node.val)
        }
        traverse(root)

        return res
    }
}
